worlds/generic/docs/RunFromSourceGuideForMac.md

```diff
@@ -0,0 +1,32 @@
+# Guide to Run Archipelago from Source Code on macOS
+Archipelago does not have a compiled release on macOS. However, it is possible to run from source code on macOS.
+## Prerequisite Software
+Here is a list of software to install and source code to download.
+1. Python 3.8 or newer from the [macOS Python downloads page](https://www.python.org/downloads/macos/).
+2. Xcode from the [macOS App Store](https://apps.apple.com/us/app/xcode/id497799835).
```

Comment:
My machine was grumping to me about not being able to run clang to install some of the python modules without a functioning instance of Xcode, so I thought Xcode was required. The grumping may have been due to having a broken instance of Xcode instead, though I'm not sure. It's been some time since I actively developed in Python.

Reply:
That grumping stopped immediately after updating my Xcode and confirming it opened properly, so that may have been some extra confirmation bias.

Reply:
`clang` definitely gets called when I first-run SNIClient.py on a Mac. I'm not sure what installed `clang` to /usr/bin/ and to /Library/Developer/CommandLineTools/usr/bin/ on my system. It's pretty clean. Conceivably it's there by default but we'd need confirmation.